### PR TITLE
DOC - RF tracking examples

### PR DESCRIPTION
**Pull request by gabknight**
This PR will address #1432.

### CODE REVIEW

**Comment by gabknight** [comment]
Waiting on #1879 and #1882 before moving on with other points of #1432.

**Comment by skoudoro** [comment]
Both has been merged, you can continue your PR 😄

**Comment by gabknight** [comment]
OK. This addresses #1432.

@skoudoro How can I build the html to test the overall display and links. `make html` fails on my machine with python 3.7

**Comment by skoudoro** [comment]
when you are in the `docs` folder, `make -C . html` should work. Let me know if it is still failing

**Comment by gabknight** [comment]
OK. thx it works, but I add to fix this unicode error.
```
dfile.write('.. AUTO-GENERATED FILE -- DO NOT EDIT!\n\n'.encode('utf-8'))
TypeError: write() argument 1 must be unicode, not str

# fix ==>
dfile.write(u'.. AUTO-GENERATED FILE -- DO NOT EDIT!\n\n')
```

**Comment by gabknight** [comment]
OK. Built examples are looking good.

- uniformized examples
- remove redundant pieces
- add figures

File: doc/examples/valid_examples.txt [hunk 1]
```diff
@@ -25,9 +24,7 @@
  segment_clustering_metrics.py
  snr_in_cc.py
  streamline_formats.py
-# tracking_eudx_odf.py
-# tracking_eudx_tensor.py
- sfm_tracking.py
+
```
Comment (skoudoro) [inline doc/examples/valid_examples.txt]: Can you remove the empty space here

File: doc/examples_index.rst [hunk 2]
```diff
@@ -145,13 +145,13 @@ Contextual enhancement
 Fiber tracking
 --------------
 
-- :ref:`example_introduction_to_basic_tracking`
```
Comment (skoudoro) [inline doc/examples_index.rst]: At the `Quick Start` section can you replace `example_tracking_quick_start` by `example_introduction_to_basic_tracking`

File: doc/examples/tracking_bootstrap_peaks.py [hunk 3]
```diff
@@ -76,24 +68,26 @@
...
 the fiber tracking (keeping all inputs the same) you will NOT get exactly the
-same set of streamlines. We can save the streamlines as a Trackvis file so it
-can be loaded into other software for visualization or further analysis.
+same set of streamlines.
 """
 
-save_trk("bootstrap_dg_CSD.trk", streamlines, affine, labels.shape)
-
 """
 Example #2: Closest peak direction getter with CSD Model
```
Comment (skoudoro) [inline doc/examples/tracking_bootstrap_peaks.py]: Can you put this text in bold or as a subtitle like this:
```
Example #2: Closest peak direction getter with CSD Model
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Can you do the same for example 1 above? thank you

File: doc/examples_index.rst (hunk 2 above)
Comment (gabknight) [reply doc/examples_index.rst]: done. example_introduction_to_basic_tracking was renamed "example_tracking_introduction_eudx"

File: doc/examples/tracking_bootstrap_peaks.py (hunk 3 above)
Comment (gabknight) [reply doc/examples/tracking_bootstrap_peaks.py]: done.

**Comment by gabknight** [comment]
@skoudoro thanks for the feedback. 

The PR is ready on my side. Let me know if you see other things to change.

**Comment by skoudoro** [comment]
Great! looks good, thank you @gabknight!  I will wait for Travis even if it doesn't run the examples and then go ahead and merge it.

**Comment by skoudoro** [comment]
Thank you for this example refactoring @gabknight ! Merging

**Pull request merged**